src/DynamicDataTable.jsx

```diff
@@ -242,9 +242,17 @@ class DynamicDataTable extends Component {
         if (props.orderByField === field.name) {
             orderByIcon = (props.orderByDirection === 'asc') ? '↓' : '↑';
         }
-
+        
+        const canOrderBy = props.orderByWhitelist.length === 0 || props.orderByWhitelist.includes(field.name);
+
+        const onClickHandler = canOrderBy
+            ? () => this.changeOrder(field)
+            : () => {};
```

Comment:
Could use a NoOp package, such as `@ninetynine/noop` for this. Would be useful in the future for when wanting to compare NoOp.